src/app/pages/signing/signing.component.ts

```diff
@@ -1,65 +1,91 @@
 import { Component, OnInit } from '@angular/core';
-
-declare var $: any;
+import { ActivatedRoute, Router } from '@angular/router';
+import { MsgIoService } from '../../core/msg-io.service';
 
 @Component({
   selector: 'app-signing',
   templateUrl: './signing.component.html',
-  styleUrls: [ './signing.component.css' ]
+  styleUrls: ['./signing.component.css']
 })
 
+/**
+ * Controls user interaction with the signing box, updating the preview display and sending the message when they
+ * click on the 'Send' button.
+ */
 export class SigningComponent implements OnInit {
+  /** Default values */
   public signingBoxText = '';
-  // fonts and colors
-  public currentFont = 'Arial';
+  public currentFont = 'Arial, sans-serif';
+  public currentFontDisplay = 'Arial';
   public backgroundColor = '#ffffff';
   public textColor = '#000000';
+  public canSend = false;
+  public charCount = 0;
+  public maxCharCount: number;
+  public charCountColor = '#bbbbbb';
+
+  private uid: string;
+  private key: string;
 
+  /**
+   * @param font - Font selected from dropdown menu
+   */
   selectFont(font: string) {
     this.currentFont = font;
+    this.currentFontDisplay = font.slice(0, font.indexOf(','));
   }
 
-  makeBold(textbox: HTMLTextAreaElement) {
+  /**
+   * The following functions add text formatting characters around selected text within textbox.
+   * @param textbox - textbox in which user types.
+   * @param formatCharacters - formatting character(s) to be placed around the selected text
+   */
+  addFormatting(textbox: HTMLTextAreaElement, formatCharacters: string) {
     const start = textbox.selectionStart;
     const end = textbox.selectionEnd;
     const text = textbox.value;
-    this.signingBoxText = text.slice(0, start) + '**' + text.slice(start, end) + '**' + text.slice(end);
+    this.signingBoxText = text.slice(0, start) + formatCharacters + text.slice(start, end) + formatCharacters + text.slice(end);
     textbox.select();
   }
 
-  makeItalics(textbox: HTMLTextAreaElement) {
-    const start = textbox.selectionStart;
-    const end = textbox.selectionEnd;
-    const text = textbox.value;
-    this.signingBoxText = text.slice(0, start) + '*' + text.slice(start, end) + '*' + text.slice(end);
-    textbox.select();
+  /**
+   * Update character count value and text colour
+   * @param textbox - textbox in which user types.
+   */
+  updateCount(textbox: HTMLTextAreaElement) {
+    this.charCount = textbox.value.length;
+    this.charCountColor = (this.charCount > this.maxCharCount ) ? '#EE1111' : '#b0b0b0';
+    this.canSend = (0 < this.charCount && this.charCount <= this.maxCharCount) ? true : false;
```

Comment:
Simplify this statement
```suggestion
    this.canSend = (0 < this.charCount && this.charCount <= this.maxCharCount);
```